R/bambu.R

```diff
@@ -153,7 +152,9 @@ bambu <- function(reads, annotations = NULL, genome = NULL, NDR = NULL,
 
     rm.readClassSe <- FALSE
     readClassList = reads
-    isBamFiles = ifelse(!is(reads, "BamFileList"), all(grepl(".bam$", reads)), FALSE)
+    isRDSs = all(sapply(reads, class)=="RangedSummarizedExperiment")
+    isBamFiles = !isRDSs
+    if(!isRDSs) isBamFiles = ifelse(!is(reads, "BamFileList"), all(grepl(".bam$", reads)), FALSE)
```

Comment:
would the following be enough to replace 156-157?
```suggestion
     isBamFiles <- all(grepl(".bam$", reads))
```

Reply:
This part of the code is very ugly, unfortunately the problem is using all() on a rangedsummarizedexperiment is very very slow which is why I check first that it isn't the loaded in .rds files.

Reply:
is it possible to provide readclasses directly? I thought it only works with bam files or a list of read class files? (not with read class objects in memory)

Reply:
(I guess the answer is that it is possible...looking through the next code, which is probably helpful to have anyway)